addons/hr_expense/models/hr_expense.py

```diff
@@ -651,15 +654,15 @@ def _parse_price(self, expense_description, currencies):
         symbols_pattern = '|'.join(symbols)
         price_pattern = "((%s)?\s?%s\s?(%s)?)" % (symbols_pattern, float_pattern, symbols_pattern)
         matches = re.findall(price_pattern, expense_description)
+        currency = currencies and currencies[0]
         if matches:
             match = max(matches, key=lambda match: len([group for group in match if group])) # get the longuest match. e.g. "2 chairs 120$" -> the price is 120$, not 2
             full_str = match[0]
             currency_str = match[1] or match[3]
             price = match[2].replace(',', '.')
 
-            if currency_str:
-                currency = currencies.filtered(lambda c: currency_str in [c.symbol, c.name])[0]
-                currency = currency or currencies[0]
+            if currency_str and currencies:
+                currency = currencies.filtered(lambda c: currency_str in [c.symbol, c.name])[0] or currency
```

Comment:
It will not work as excepted as `[0]` raises an exception on an empty recordset :-)

Reply:
@kba-odoo Indeed, my bad, thanks!
You prefer something like what just re-pushed or this:
```py
 if currency_str and currencies:
                currencies = currencies.filtered(lambda c: currency_str in [c.symbol, c.name])
                currency = currencies and currencies[0]
```

Reply:
> @kba-odoo Indeed, my bad, thanks! You prefer something like what I just re-pushed or something like this:
> 
> ```python
> if currency_str and currencies:
>     currency = currencies.filtered(lambda c: currency_str in [c.symbol, c.name])
>     currency = currency or currencies[0]
> ```

Maybe more something like this? :-)

```python
if currency_str and currencies:
    currency = currencies.filtered(lambda c: currency_str in [c.symbol, c.name])
    currency = (currencies and currencies[0]) or currency
```

Reply:
I guess you meant `currencies = currencies.filtered(...)` :wink: 
Updated :+1: